### PR TITLE
refactor: implement new plugin interfaces

### PR DESCRIPTION
**Pull request by mindlesscloud**
# Summary
 implement new plugin interfaces
### Key Points

- [ ] This is a breaking change
- [ ] New or existing documentation is updated

### Description
Describe what this PR does, and aims to solve in a few sentences.

### Does this close any open issues?
Please mention the issues here.

### Current Behavior
Describe the current behaviour of this issue, if relevant.

### New Behavior
Describe the new behaviour updated in this issue, if relevant.

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/jira/jira.go [hunk 1]
```diff
@@ -445,63 +529,18 @@ var PluginEntry Jira //nolint
...
-	boardId, err := strconv.ParseUint(args[1], 10, 64)
-	if err != nil {
-		panic(fmt.Errorf("error paring board_id: %w", err))
-	}
-
-	err = core.RegisterPlugin("jira", PluginEntry)
+	cmd := &cobra.Command{Use: "jira"}
+	sourceId := cmd.Flags().Uint64P("source", "s", 0, "jira source id")
+	boardId := cmd.Flags().Uint64P("board", "b", 0, "jira board id")
+	err := cmd.MarkFlagRequired("project-id")
```
Comment (warren830) [inline plugins/jira/jira.go:535]: should require source and board, not project-id

Comment (mindlesscloud) [reply plugins/jira/jira.go:535]: oops

File: plugins/jira/tasks/sprint_converter.go [hunk 2]
```diff
@@ -31,12 +30,12 @@ func ConvertSprints(taskCtx core.SubTaskContext) error {
 		return err
 	}
 	defer cursor.Close()
-
+	var converter *helper.DataConverter
```
Comment (warren830) [inline plugins/jira/tasks/sprint_converter.go:33]: why do we need to define like this here

Comment (mindlesscloud) [reply plugins/jira/tasks/sprint_converter.go:33]: to fix a linter warning regarding potential variable shadowing

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**